Have DataGrip configure it using "Advanced Settings"

QiE2035 opened the pull request:
[Bing translated from Chinese]

Add some `DriverPropertyInfo` to `getPropertyInfo` so that DataGrip *(IntelliJ IDEA, CLion, etc)* can be configured with "Advanced Settings".

I'll admit that this implementation is really bad, but at least now we have an "Advanced Settings" with some content.

![image](https://github.com/h2database/h2database/assets/18079122/e11eb1cc-2118-4c62-b1b3-788441592cd9)

QiE2035 committed:
Add some `DriverPropertyInfo` to `getPropertyInfo` so that DataGrip can be configured with "Advanced Settings"

QiE2035 committed:
Fix "MODE" default `null`

QiE2035 committed:
Move array to static const.

katzyn commented on h2/src/main/org/h2/util/DriverPropertyInfoSet.java:
```diff
@@ -0,0 +1,78 @@
+package org.h2.util;
```
This class should be in `org.h2.engine` package to avoid interaction with actual configuration classes.

katzyn commented on h2/src/main/org/h2/util/DriverPropertyInfoSet.java:
```diff
@@ -0,0 +1,79 @@
+            newProp("RECOMPILE_ALWAYS", false, "Database setting RECOMPILE_ALWAYS (default: false). \nAlways recompile prepared statements."),
+            newProp("REUSE_SPACE", true, "Database setting REUSE_SPACE (default: true). \nIf disabled, all changes are appended to the database file, and existing content is never overwritten. \nThis setting has no effect if the database is already open."),
+            newProp("SHARE_LINKED_CONNECTIONS", true, "Database setting SHARE_LINKED_CONNECTIONS (default: true). \nLinked connections should be shared, that means connections to the same database should be used for all linked tables that connect to the same database."),
+            newProp("ZERO_BASED_ENUMS", false, "Database setting ZERO_BASED_ENUMS (default: false). \nIf set, ENUM ordinal values are 0-based."),
```
- There is no `MV_STORE` setting.
- `ZERO_BASED_ENUMS` is going to be removed, don't list it here too.
- I think it's a bad idea to hard-code defaults here, it will be better to grab actual defaults from `DbSettings` or other sources.
- Many settings here (`OPTIMIZE_*` and possibly others) are just kill switches. If some bug in optimization will be found, such switch can be used as a temporary workaround until this bug will be fixed. I think it is a bad idea to return them here, because they can be misused by accident.
- `DEFAULT_TABLE_ENGINE` shouldn't be listed too, it exists for exotic use cases in third-party solutions.

I think it will be better to list only useful settings here, including some settings from other sources, take a look on `ConnectionInfo`, it contains names of various other settings, including popular ones. @grandinj, what do you think?

You also need to break long lines, our code formatting rules don't allow lines longer than 120 characters.

katzyn commented on h2/src/main/org/h2/Driver.java:
```diff
@@ -117,7 +118,7 @@ public int getMinorVersion() {
     @Override
     public DriverPropertyInfo[] getPropertyInfo(String url, Properties info) {
-        return new DriverPropertyInfo[0];
+        return DriverPropertyInfoSet.get();
```
`TestMetaData` needs to be updated.

katzyn commented on h2/src/main/org/h2/util/DriverPropertyInfoSet.java:
```diff
@@ -0,0 +1,79 @@
+            final String name,
+            final String value,
+            final String desc,
+            final String... choice) {
```
Usually we don't mark method parameters and local variables as `final`.

katzyn reviewed (commented):
Thank you for your contribution!

QiE2035 commented:
Ok, so do I need to modify it right away? I seem to see you initiating a discussion with another person in the comments, do you need me to wait for the outcome of your discussion?

katzyn commented:
Which settings we want to announce is an open question, but you can try to resolve other problems right now.

QiE2035 committed:
Fix DriverPropertyInfoSet

1. Move to `org.h2.engine`
2. Rename to `DriverPropertyInfoProvider`
3. Break long lines
4. Comment unnecessary property
5. Use `DbSettings.DEFAULT` as default values
